### DISCUSSION TITLE
Tag filter discussion

### DISCUSSION THREAD

**Discussion by Quafadas** [lefou/mill-vcs-version#154] [Ideas]
I have a mill project, for which I would like manual control of doc publishing - i.e. only publish the docs on some manual trigger. 

My initial instinct was to have a "doc" tag in a similar vein to "v0.0.1" tags, and have the CI filter for when it's found. However, this messes with the vcsVersion plugin (it detects "doc" as the last tag, because it is the last tag). Before I go and invent something dumb, has anyone thought this one through before :-)... ? 

> Tobias Roeser — Today at 09:33
> Maybe, we want to add a tag filter or a tag mapper to vcs-version plugin.

If you feel this could indeed belong in the plugin, I would try to PR it. 

I had a think on what that could look like. I could imagine changing the signature of 

```def vcsState: Input[VcsState] = T.input { calcVcsState(T.log) }```

to something like 

```
def vcsState(filter: Option[String] = None): Input[VcsState] = T.input { calcVcsState(T.log) }
```
But I don't really want to mess with the core parts of the plugin too much, so I'd be hesitant.

The other idea, would be to introduce a new task - something like 
```
def lastTagMatching(filter: Option[String] = None): Input[VcsState] = T.input { gitFilterTags(T.log) }
```
Do you think either of these could be viable?

**Comment by Quafadas** [comment]
After a little research, I realised it's possible to trigger workflows via a manual trigger in the UI. 

```yaml
on:
  workflow_dispatch:  # Manual trigger
```

I think this is a better solution so I'm going to go with that for the time being ...

**Comment by lefou** [comment]
My initial understanding was, that you need to ignore some specific tags, so that you get a nice version numbers instead of "docs". Might be doable. We need to come up with a different configuration thought, as input tasks can't accept arguments.

**Reply by Quafadas** [reply]
You are right - my original idea was to use tags for releasing _both_ docs and artifacts with tags. That would require this mechanism, as there would be two “series” of tags, stepping on each others toes.

However, the requirement really, is only to be able to trigger a release of the docs manually.

I spent some time researching  and I honestly think a manual trigger in the GHA gui is better suited to this particular use case. 

I would implement this for interest if you think it is valuable, I would wish to support the mill ecosystem where I am able. I don’t think this feature would be  the right thing for this particular problem though. I do apologise for the spam :-/…
